### PR TITLE
Change `$TAG` to `$DIR` in `execute-gcc-tests.sh` and `create-gcc-summary.sh` so it is easier to use them standalone

### PR DESCRIPTION
**Pull request by Blackhex**
This allows re-using the scripts in https://github.com/Windows-on-ARM-Experiments/msys2-woarm64-build/pull/15.

Tested by https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build/actions/runs/10697695783

### CODE REVIEW

File: .github/workflows/build-and-test-toolchain.yml [hunk 1]
```diff
@@ -141,13 +141,14 @@ jobs:
       - name: Execute GCC tests
         run: |
           cd ~/work
-          .github/scripts/toolchain/execute-gcc-tests.sh "${{ env.TAG }}" "${{ env.MODULE }}" "${{ env.FILTER }}"
+          .github/scripts/toolchain/execute-gcc-tests.sh $ARTIFACT_PATH/gcc-tests-${{ env.TAG }} \
```
Comment (eukarpov) [inline .github/workflows/build-and-test-toolchain.yml:144]: could it be refactored with ~/work/github... without `cd` call?

**Review by eukarpov — APPROVED** [review]

File: .github/workflows/build-and-test-toolchain.yml (hunk 1 above)
Comment (Blackhex) [reply .github/workflows/build-and-test-toolchain.yml:144]: OK, I'll do a generic refactor in a separate PR.

**Comment by Blackhex** [comment]
Changed based on discussion at https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build/pull/188/files/763c14fa6599efffb359f30d5d15b4d1df997a24#r1741559151

**Comment by vejbomar** [comment]
Can you do the same refactor also for `.github/scripts/toolchain/group-gcc-test-failures.sh` to keep them unified?

**Comment by Blackhex** [comment]
> Can you do the same refactor also for `.github/scripts/toolchain/group-gcc-test-failures.sh` to keep them unified?

Good point, thank you for noticing.

File: .github/scripts/toolchain/create-gcc-summary.sh [hunk 2]
```diff
@@ -47,7 +47,7 @@ TOTAL_UNSUPPORTED_TESTS=0
 TOTAL_DEJAGNU_ERRORS=0
 
 set +x # echo off
-for SUM_FILE in "$ARTIFACT_PATH/gcc-tests-$TAG/"*.sum; do
+for FILE in `find $DIR -path '*.sum'`; do
```
Comment (vejbomar) [inline .github/scripts/toolchain/create-gcc-summary.sh:50]: Why is the `find` needed? Is it for some usecase outside CI?

**Comment by vejbomar** [comment]
Beside one more question about `find` this looks good to me. Thanks for simplifying summary steps into one, that is a lot easier to understand now imo since all steps stay in WSL bash instead of switching to powershell

File: .github/scripts/toolchain/create-gcc-summary.sh (hunk 2 above)
Comment (Blackhex) [reply .github/scripts/toolchain/create-gcc-summary.sh:50]: To collect results directly from GCC souce code where `*.sum` files are in different subfolders after running `make check`.

**Review by vejbomar — APPROVED** [review]

**Review by eukarpov — APPROVED** [review]

File: .github/workflows/build-and-test-toolchain.yml [hunk 3]
```diff
@@ -113,14 +113,15 @@ jobs:
...
-          echo "artifact-path=`wslpath -w $ARTIFACT_PATH`" >> $GITHUB_OUTPUT
+          echo "build-path-win=`wslpath -w $BUILD_PATH`" >> $GITHUB_OUTPUT
+          echo "ccache-dir-path-win=`wslpath -w $CCACHE_DIR_PATH`" >> $GITHUB_OUTPUT
+          echo "artifact-path-win=`wslpath -w $ARTIFACT_PATH`" >> $GITHUB_OUTPUT
+          echo "artifact-path-wsl=$ARTIFACT_PATH" >> $GITHUB_OUTPUT
 
       - name: Restore Ccache
         uses: actions/cache/restore@v4
         with:
-          path: ${{ steps.get-wsl-paths.outputs.ccache-dir-path }}
```
Comment (johnandbrandimelton25-lgtm) [inline .github/workflows/build-and-test-toolchain.yml:123]: - [ ]

**Pull request merged**